### PR TITLE
Issue/filtering on non existent gives 400

### PR DESCRIPTION
**Pull request by stevenbal**
Filtering on URL fields now has URL validation and filtering on a valid url that does not point to the correct resource simply returns an empty list

### CODE REVIEW

**Comment by stevenbal** [comment]
> Can you please add test case for `URLModelChoiceField.to_python` method?

I have added two tests for it, thanks for the feedback :)

File: vng_api_common/filters.py [hunk 1]
```diff
@@ -73,6 +74,10 @@ def __init__(self, *args, **kwargs):
     def url_to_pk(self, url: str):
         parsed = urlparse(url)
         path = parsed.path
+
+        if path == "/":
+            path = ""
```
Comment (sergei-maertens) [inline vng_api_common/filters.py]: Why is this needed?

File: vng_api_common/filters.py [hunk 2]
```diff
@@ -86,14 +91,17 @@ def url_to_pk(self, url: str):
         return instance.pk
 
     def to_python(self, value: str):
-        # TODO: validate that it's proper URL input
+        if value is not None:
+            try:
+                URLValidator()(value)
+            except ValidationError as exc:
+                raise exc
```
Comment (sergei-maertens) [inline vng_api_common/filters.py]:
```python
if value is not None:
    URLValidator()(value)
```
has the same end-result

File: vng_api_common/filters.py [hunk 3]
```diff
@@ -86,14 +91,17 @@ def url_to_pk(self, url: str):
...
+
         if value:
             try:
                 value = self.url_to_pk(value)
             except models.ObjectDoesNotExist:
-                raise ValidationError(
-                    _("Invalid resource URL supplied"), code="invalid"
-                )
+                return []
```
Comment (sergei-maertens) [inline vng_api_common/filters.py]: Can you please add a `logger.info(...)` call? We will definitely get support questions about this.

File: vng_api_common/filters.py (hunk 1 above)
Comment (stevenbal) [reply vng_api_common/filters.py]: if I would try to filter on for instance `?catalogus=https://example.com/`, the path would be `"/"`, and this does not raise an `ObjectDoesNotExist`, but it when checking if the callback has a viewset (here: https://github.com/VNG-Realisatie/vng-api-common/blob/master/vng_api_common/utils.py#L61), causing an internal server error

I guess it is better to remove the trailing slash in `get_resource_for_path`
